packages/eslint-plugin/tests/rules/switch-exhaustiveness-check.test.ts

```diff
@@ -209,6 +209,22 @@ function test(value: ObjectUnion): number {
   }
 }
     `,
+    // switch with default clause on non-union type
+    {
+      code: `
+const value: number = Math.floor(Math.random() * 3);
```

Comment:
[Cleanup] Just nitpicking, what you're looking to do is more succinctly described by:

```suggestion
declare const value: number;
```

Not a blocking request if you hate this 😄